### PR TITLE
Update README.md for new build.

### PR DESCRIPTION
**Pull request by awong-dev**
The new Metalsmith based build reorganized a bunch of files and moved the system away from Ruby to Node.js. This updates the README to reflect the changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,93 +1,311 @@
...
+## I want to...
+
+| I want to... | Then you should... |
+| ------------ | ------------------ |
+| check out the site | `git clone https://github.com/department-of-veterans-affairs/vets-website.git` followed by `npm install`. Run `npm install` anytime `package.json` changes. | 
+| deploy the site | merge to master for `dev.vets.gov` and `staging.vets.gov`. Merge to production for `www.vets.gov`. Travis will do the deploy on the post merge build. Submit a trivial change to force a re-deploy. |
+| update static content that is already on the site. | Find the corresponding file in `content/pages`. Make your edit. Send a PR. |
+| add new static content to the site. | Create new files at the right location in `content/pages`. Send a PR. |
+| build the site with dev features enabled. | `npm run build` |
+| build the production site (dev features disabled). | `npm run build -- --buildtype production` |
```
Comment (plusjeff) [inline README.md]: Is there an extra `--` here?

File: README.md [hunk 2]
```diff
@@ -1,93 +1,311 @@
...
+documentation around React is based on a Webpack toolchain.
 
 ## More documentation
 
 - [Why Is My Build Breaking?](docs/WhyIsMyBuildBreaking.md)
 - [How Breadcrumbs Work](docs/HowBreadCrumbsWork.md)
 - [How URLs are Created](docs/HowURLsAreCreated.md)
-- [How Search Works](docs/HowSearchWorks.md)
\ No newline at end of file
+- [How Search Works](docs/HowSearchWorks.md)
```
Comment (plusjeff) [inline README.md]: Are these still relevant?

**Comment by plusjeff** [comment]
This is great! One question I'd still love the README to answer explicitly is: How do I add a new React "app"/section/etc (i.e. what does the file structure in `src` need to be, what needs to be set up in terms of routes so that it's navigable, what needs to be added in the content area, or anything else that has to be done).

**Comment by awong-dev** [comment]
Good point.  I will add in followup PR since this one is already huge.

File: README.md (hunk 1 above)
Comment (awong-dev) [inline README.md]: nope. That's intentional. Will add a note.

File: README.md (hunk 2 above)
Comment (awong-dev) [inline README.md]: Dunno...will add TODO.

**Comment by plusjeff** [comment]
LGTM

**Pull request merged**